### ISSUE TITLE
The "d" alias on the "Directory" parameter for Get-ChildItem can never work

### ISSUE THREAD

**Issue by Jaykul** [PowerShell/PowerShell#14700]
If you look at the parameters of Get-ChildItem, there's an alias:

```PowerShell
> (Get-Command Get-ChildItem).Parameters["Directory"].Aliases
ad
d
```

The problem is that the `d` alias **CANNOT** ever work, because the `Directory` parameter is dynamic, and `Get-ChildItem` already has a parameter that starts with "d" (the "Depth" parameter), so the `-d` switch always gets bound to the `-Depth` parameter (and the user gets prompted for an actual value).

Can we remove that alias, since it's just confusing users?

**Comment by mklement0** [comment]
Good point, previously reported in #4739 (I've just closed the latter, because this issue is more focused).

Given that binding to the _static_ parameters seemingly always takes precedence, the lessons to be drawn for future cmdlets are:

* Do not define aliases for _dynamic_ parameters that are prefix matches of the _static_ parameter names / aliases, because the latter will preempt the former.

  * _Ideally_ - I don't know if it's feasible, given that it could only be detected at runtime - cmdlets would detect this condition and report an error.

* Static parameters taking precedence makes sense conceptually, especially in the context of provider cmdlets, so that, say, `-d` is _always_ `-Depth`, irrespective of what provider is targeted, and not _situationally_ either `-Depth`  or `-Directory`.

**Comment by daxian-dbw** [comment]
Closed via #15171
